src/core/grid_based_algorithms/lbgpu_cuda.cu

```diff
@@ -2307,12 +2246,13 @@ __global__ void
 calc_fluid_particle_ia(LB_nodes_gpu n_a, CUDA_particle_data *particle_data,
                        float *particle_force, LB_node_force_density_gpu node_f,
                        LB_rho_v_gpu *d_v, bool couple_virtual,
-                       uint64_t philox_counter, float friction) {
+                       uint64_t philox_counter, float friction,
+                       float *lb_boundary_velocity, bool three_point_coupling) {
```

Comment:
Since there are only two coupling schemes, and this code is performance relevant, I think the coupling scheme should be selected statically and code for both schemes without the branches should be generated. E.g. make the coupling scheme a template parameter and instantiate both versions of this kernel.